### ISSUE TITLE
Failing build with HPX without SEPARATE_LIBS

### ISSUE THREAD

**Issue by masterleinad** [kokkos/kokkos#2453]
We currently don't link to `HPX` (in case the backend is enabled) when building the object libraries. When they are proper libraries this works though. Hence, this is also fixed by #2447.

**Comment by jjwilke** [comment]
Can you post what the actual issue is for future reference?

**Comment by masterleinad** [comment]
Sure. With `CMake 3.10.2` I get
```
cd /home/darndt/kokkos/build/containers/src && /usr/bin/g++   -I/home/darndt/kokkos/build/containers/src -I/home/darndt/kokkos/containers/src -I/home/darndt/kokkos/build -I/home/darndt/kokkos/build/core/src -I/home/darndt/kokkos/core/src  -std=gnu++14 -o CMakeFiles/kokkoscontainers.dir/impl/Kokkos_UnorderedMap_impl.cpp.o -c /home/darndt/kokkos/containers/src/impl/Kokkos_UnorderedMap_impl.cpp
In file included from /home/darndt/kokkos/core/src/Kokkos_HPX.hpp:60:0,
                 from /home/darndt/kokkos/core/src/Kokkos_Core.hpp:70,
                 from /home/darndt/kokkos/containers/src/Kokkos_UnorderedMap.hpp:53,
                 from /home/darndt/kokkos/containers/src/impl/Kokkos_UnorderedMap_impl.cpp:44:
/home/darndt/kokkos/core/src/HPX/Kokkos_HPX_ChunkedRoundRobinExecutor.hpp:47:10: fatal error: hpx/config.hpp: No such file or directory
 #include <hpx/config.hpp>
          ^~~~~~~~~~~~~~~~
compilation terminated.
```
due to: https://github.com/kokkos/kokkos/blob/904b463d6b79e72135210efe06f0d86942ed1d4e/cmake/kokkos_tribits.cmake#L358-L376 We simply don't add the appropriate include directories in that case.

**Comment by masterleinad** [comment]
Addressed in #2447.
